src/components/CounterValuePolling.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+import React from "react";
+import { AppState } from "react-native";
+import hoistNonReactStatic from "hoist-non-react-statics";
+import throttle from "lodash/throttle";
+import { pollRates } from "../actions/counterValues";
+
+// $FlowFixMe
+export const PollingContext = React.createContext(() => {});
+
+export type CounterValuePolling = {
+  poll: () => Promise<*>,
```

Comment:
I'm not sure why `poll` function should be in `state`. It's assigned `this.poll` and is never reassigned

Reply:
yeah, in this component, `this.state` is that value that is provided by context.
if i'm not wrong, in the new context API, you should provide an object that is equal by reference `===` & not creating it each time. `this.state` have this property so it was win-win to use this.state as context value... you can see i did the same for LocaleContext https://github.com/LedgerHQ/ledger-wallet-mobile/blob/master/src/components/LocaleContext.js i'm not sure if it's a pattern found/approved by React team but i'll try to discuss on this subject later with them.

Reply:
note that CounterValuePolling type, is the type that is exposed by the context, before being a state ;) it could live somewhere else technically, it's just convenient like this.

Reply:
makes sense